tests/test_repository_tool.py

```diff
@@ -369,6 +369,108 @@ def test_writeall(self):
     self.assertRaises(securesystemslib.exceptions.FormatError, repository.writeall, 3)
 
 
+  def test_writeall_no_files(self):
```

Comment:
Just wanted to give kudos for adding a test case that closely mirrors the current work on Warehouse ❤️